src/utils/displayFunctions.ts

```diff
@@ -70,12 +69,22 @@ export const makeDisplayFunctions = (brandToInfo: Map<Brand, BrandInfo>) => {
     getLogoForBrandPetname(getPetname(brand));
 
   const displayPrice = (price: PriceDescription) => {
-    const givenUnitsOfBrandIn = price.amountIn.value / PRICE_BRAND_UNIT_AMOUNT;
-    const pricePerUnitOfBrandIn = price.amountOut.value / givenUnitsOfBrandIn;
-    return (
-      '$' +
-      stringifyValue(pricePerUnitOfBrandIn, AssetKind.NAT, USD_BRAND_DECIMALS)
+    const { amountIn, amountOut } = price;
+    const { brand: brandIn } = amountIn;
+    const brandInDecimals = getDecimalPlaces(brandIn);
+    assert(brandInDecimals);
+
+    const unitAmountOfBrandIn = AmountMath.make(
+      brandIn,
+      10n ** BigInt(brandInDecimals),
     );
+
+    const brandOutAmountPerUnitOfBrandIn = floorMultiplyBy(
+      unitAmountOfBrandIn,
+      makeRatioFromAmounts(amountOut, amountIn),
+    );
+
+    return '$' + displayAmount(brandOutAmountPerUnitOfBrandIn);
```

Comment:
this assumes `brandOut` is IST?

Reply:
Yes, will it ever not be?

Reply:
discussed in https://github.com/Agoric/dapp-inter/pull/41#discussion_r1090053754 . resolving this thread.